backend/fbdev.c

```diff
@@ -59,7 +61,23 @@ static void _twin_fbdev_put_span(twin_coord_t left,
     off_t off = top * screen->width + left;
     uint32_t *dest =
         (uint32_t *) ((uintptr_t) tx->fb_base + (off * sizeof(*dest)));
-    memcpy(dest, pixels, width * sizeof(*dest));
+
+    /* If the framebuffer is in 16 bpp mode, convert each pixel from 32 bpp (ARGB) to 16 bpp (RGB565).
+    * - Extracts the red, green, and blue channels from the 32-bit color value.
+    * - Shifts and masks each channel to fit into the 16-bit RGB565 format:
+    *      - 5 bits for red
+    *      - 6 bits for green
+    *      - 5 bits for blue
+    * - Combines the shifted color values into a single 16-bit pixel.
+    * If the framebuffer is in 32 bpp mode, directly copy the pixels to the framebuffer.
+    */
+    if(tx->fb_var.bits_per_pixel == 16){
```

Comment:
Since `_twin_fbdev_put_span` is called frequently, you would avoid the branches as possible. You can register `_twin_fbdev_put_span16` or `_twin_fbdev_put_span24` respectively.

Reply:
I found that when `_twin_fbdev_put_span` is called frequently, the following code is repeatedly checked, which leads to a performance degradation.
```c
static void _twin_fbdev_put_span(twin_coord_t left,
                                 twin_coord_t top,
                                 twin_coord_t right,
                                 twin_argb32_t *pixels,
                                 void *closure)
{
...
    if (tx->fb_base == MAP_FAILED)
        return;
...
}
```

Reply:
> I noticed that `_twin_fbdev_put_span16`, `_twin_fbdev_put_span24`, and `_twin_fbdev_put_span32` contain duplicate code. I want to optimize the redundant code by using `inline`.

Alternatively, use macro to maintain a template for compile-time code generation.

Reply:
> I found that when _twin_fbdev_put_span is called frequently, the condition `tx->fb_base == MAP_FAILED` is repeatedly checked, which leads to a performance degradation.

@shengwen-tw, Can you confirm this?